server/odc-service/src/main/java/com/oceanbase/odc/service/integration/saml/AddableRelyingPartyRegistrationRepository.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.integration.saml;
+
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistration;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrationRepository;
+import org.springframework.stereotype.Component;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.oceanbase.odc.core.shared.Verify;
+import com.oceanbase.odc.service.integration.IntegrationService;
+import com.oceanbase.odc.service.integration.model.SSOIntegrationConfig;
+
+@Component
+public class AddableRelyingPartyRegistrationRepository implements RelyingPartyRegistrationRepository {
+
+    public final Cache<String, SSOIntegrationConfig> testConfigRegistrations =
```

Comment:
Why do we need a cache here?

Reply:
For test login, test login config not require persist.